packages/react-components/react-tabster/src/hooks/useTabster.ts

```diff
@@ -1,7 +1,26 @@
 import * as React from 'react';
 import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
-import { createTabster, disposeTabster, Types as TabsterTypes } from 'tabster';
-import { useIsomorphicLayoutEffect, getParent } from '@fluentui/react-utilities';
+import { createTabster, disposeTabster, getShadowDOMAPI, Types as TabsterTypes } from 'tabster';
+import { useIsomorphicLayoutEffect, getParent, getParentInShadowDOM } from '@fluentui/react-utilities';
+
+interface WindowWithTabsterShadowDOMAPI extends Window {
+  __tabsterShadowDOMAPI?: TabsterTypes.DOMAPI;
+}
+
+/**
+ * Imports support for ShadowDOM and assigns it to the current window, so that it can be used from useTabster()
+ * but remains tree-shakeable when not needed.
+ *
+ * Must be called before useTabster() is called, right after FluentProvider.
+ *
+ * @returns Tabster ShadowDOM API
+ */
+export const useShadowDOMSupport = (): TabsterTypes.DOMAPI | undefined => {
+  const { targetDocument } = useFluent();
+  const defaultView = targetDocument?.defaultView as WindowWithTabsterShadowDOMAPI | undefined;
+
+  return defaultView ? (defaultView.__tabsterShadowDOMAPI = getShadowDOMAPI()) : undefined;
```

Comment:
I like this idea! So far we've been keeping all the shadow DOM features out of core and in `@fluenui-contrib`. I think this hook could live in contrib and then `createTabster()` could be updated to:

1. Take the param as you have it below
2. Or, internally, look up `__tabsterShadowDOMAPI` on window if the param is not provided

In this way we wouldn't have to make any updates to Fluent Core for this feature and would still get this behavior.

WDYT?

Reply:
Can we remove this now that the hooks lives in contrib?

Reply:
This is the old draft PR, I will abandon it. The correct one is here: https://github.com/microsoft/fluentui/pull/30429

Reply:
Got it. Thanks!